pallets/schemas/src/lib.rs

```diff
@@ -381,7 +422,49 @@ pub mod pallet {
 		) -> DispatchResult {
 			let sender = ensure_signed(origin)?;
 
-			let schema_id = Self::create_schema_for(model, model_type, payload_location, settings)?;
+			let schema_id =
+				Self::create_schema_for(model, model_type, payload_location, settings, None)?;
+
+			Self::deposit_event(Event::SchemaCreated { key: sender, schema_id });
+			Ok(())
+		}
+
+		/// Adds a given schema to storage. The schema in question must be of length
+		/// between the min and max model size allowed for schemas (see pallet
+		/// constants above). If the pallet's maximum schema limit has been
+		/// fulfilled by the time this extrinsic is called, a SchemaCountOverflow error
+		/// will be thrown.
+		///
+		/// # Events
+		/// * [`Event::SchemaCreated`]
+		///
+		/// # Errors
+		/// * [`Error::LessThanMinSchemaModelBytes`] - The schema's length is less than the minimum schema length
+		/// * [`Error::ExceedsMaxSchemaModelBytes`] - The schema's length is greater than the maximum schema length
+		/// * [`Error::InvalidSchema`] - Schema is malformed in some way
+		/// * [`Error::SchemaCountOverflow`] - The schema count has exceeded its bounds
+		/// * [`Error::InvalidSetting`] - Invalid setting is provided
+		///
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::WeightInfo::create_schema_v3(model.len() as u32 + settings.len() as u32))]
+		pub fn create_schema_v3(
+			origin: OriginFor<T>,
+			schema_name: SchemaNamePayload,
```

Comment:
Question: use combined name like this or maybe have two separate params for namespace and descriptor?